### PR TITLE
vmauth: fix requests routing by host when using `src_hosts`

### PR DESCRIPTION
**Pull request by Haleygo**
Most requests use the [origin form](https://datatracker.ietf.org/doc/html/rfc7230#section-5.3) rather than the absolute form. So in [Request.URL](https://pkg.go.dev/net/http#Request), fields other than Path and RawQuery will be empty.
> 	// For server requests, the URL is parsed from the URI
> 	// supplied on the Request-Line as stored in RequestURI.  For
> 	// most requests, fields other than Path and RawQuery will be
> 	// empty. (See RFC 7230, Section 5.3)

Using `Host` header instead to match `src_hosts`. 
It also allows user to route requests with customized `Host` header.

### CODE REVIEW

File: docs/changelog/CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ Released at 2024-11-29
 * BUGFIX: `vmselect` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): properly handle [multitenant](https://docs.victoriametrics.com/cluster-victoriametrics/#multitenancy-via-labels) query request errors and correctly perform search for available tenants. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7549) for details. This is follow-up after [v1.106.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.106.1) release changes.
 * BUGFIX: `vmselect` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): properly set `multitenant` cache expiration duration with `search.tenantCacheExpireDuration`. Previously flag always used default value.
 * BUGFIX: [vmauth](https://docs.victoriametrics.com/vmauth/): properly init `ip_filters` for `unauthorized_user` config section. Previously it was ignored and `vmauth` didn't apply `ip_filter` for `unauthorized` access.
+* BUGFIX: [vmauth](https://docs.victoriametrics.com/vmauth/): fix requests routing by host when using `src_hosts`. Previously, request header could be ignored.
```
Comment (zekker6) [inline docs/changelog/CHANGELOG.md:44]: Looks like changelog needs to be moved to `tip` section.

**Review by zekker6 — APPROVED** [review]

File: docs/changelog/CHANGELOG.md (hunk 1 above)
Comment (Haleygo) [reply docs/changelog/CHANGELOG.md]: yep, rebased, thanks

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**